source/solvers/navier_stokes_base.cc

```diff
@@ -2483,6 +2483,102 @@ NavierStokesBase<dim, VectorType, DofsType>::
+  const unsigned int display_precision)
+{
+  if constexpr (std::is_same_v<VectorType, GlobalVectorType> ||
+                std::is_same_v<VectorType,
+                               LinearAlgebra::distributed::Vector<double>>)
+    {
+      FEValuesExtractors::Vector velocities(0);
+      FEValuesExtractors::Scalar pressure(dim);
+
+      ComponentMask velocity_mask = fe->component_mask(velocities);
+      ComponentMask pressure_mask = fe->component_mask(pressure);
+
+      const std::vector<IndexSet> index_set_velocity =
+        DoFTools::locally_owned_dofs_per_component(dof_handler, velocity_mask);
+      const std::vector<IndexSet> index_set_pressure =
+        DoFTools::locally_owned_dofs_per_component(dof_handler, pressure_mask);
+
+      double local_sum = 0.0;
+      double local_max = DBL_MIN;
+
+      for (unsigned int d = 0; d < dim; ++d)
+        {
+          for (auto j = index_set_velocity[d].begin();
+               j != index_set_velocity[d].end();
+               j++)
+            {
+              double dof_newton_update = newton_update[*j];
+
+              local_sum += dof_newton_update * dof_newton_update;
+
+              if (dof_newton_update > local_max)
+                {
+                  local_max = dof_newton_update;
+                }
+            }
+        }
+
+      double global_velocity_l2_norm =
+        std::sqrt(Utilities::MPI::sum(local_sum, this->mpi_communicator));
+      double global_velocity_linfty_norm =
+        Utilities::MPI::max(local_max, this->mpi_communicator);
+
+      local_sum = 0.0;
+      local_max = DBL_MIN;
+
+      for (auto j = index_set_pressure[dim].begin();
+           j != index_set_pressure[dim].end();
+           j++)
+        {
+          double dof_newton_update = newton_update[*j];
+
+          local_sum += dof_newton_update * dof_newton_update;
+
+          if (dof_newton_update > local_max)
+            {
+              local_max = dof_newton_update;
+            }
+        }
+
+      double global_pressure_l2_norm =
+        std::sqrt(Utilities::MPI::sum(local_sum, this->mpi_communicator));
+      double global_pressure_linfty_norm =
+        Utilities::MPI::max(local_max, this->mpi_communicator);
+
+      this->pcout << std::setprecision(display_precision)
+                  << "\n\t||du||_L2 = " << std::setw(6)
+                  << global_velocity_l2_norm << std::setw(6)
+                  << "\t||du||_Linfty = "
+                  << std::setprecision(display_precision)
+                  << global_velocity_linfty_norm << std::endl;
+      this->pcout << std::setprecision(display_precision)
+                  << "\t||dp||_L2 = " << std::setw(6) << global_pressure_l2_norm
+                  << std::setw(6) << "\t||dp||_Linfty = "
+                  << std::setprecision(display_precision)
+                  << global_pressure_linfty_norm << std::endl;
+    }
+  if constexpr (std::is_same_v<VectorType, GlobalBlockVectorType>)
```

Comment:
```suggestion
  else if constexpr (std::is_same_v<VectorType, GlobalBlockVectorType>)
```

Reply:
I don't see the point of the else here.  it's actually better to have an undefined behavior than have an else taht could maybe not be an else (other than if you throw at the end something in an else)